README-SEMANTICS.md

```diff
@@ -0,0 +1,248 @@
+
+## Update Event
+
+The user can update existing events. For example, if
+the user says "Chang my meeting tomorrow to 4 PM", then the program is:
+```clojure
+(Yield :output 
+  (UpdateCommitEventWrapper :event 
+    (UpdatePreflightEventWrapper 
+      :id 
+        (:id 
+          (singleton 
+            (:results 
+              (FindEventWrapperWithDefaults :constraint
+                (Constraint[Event] :start 
+                  (Constraint[DateTime] :date (?= (Tomorrow)))) 
+                )
+              )
+            )
+          )
+        )
+      :update 
+        (Constraint[Event] :start 
+          (Constraint[DateTime] :time 
+            (?= (NumberPM :number #(Number 4)))
+          )
+        )
+    )
+  )
+)
+```
+In this program, `UpdatePreflightEventWrapper` takes two
+arguments: `id` and `update`. The `id` argument specifies the event id
+to be updated. It is obtained through querying the database
+via a constraint, which requires the event's date to be tomorrow.
+The `update` argument encodes the requirements for the update. Here,
+it requires the new event's start time to be at 4 PM. The
+`UpdatePreflightEventWrapper` function uses the `id` to extract an
+event from the database, then it uses the `update` constraint to revise it.
+The revision guarantees that the new event will satisfy the user constraint, but
+preserving the original properties as long as they don't
+conflict with the user constraint. For example, an "1-hour
+brainstorm meeting at 3 PM" will be revised to be an "1-hour
+brainstorm meeting at 4 PM".
+
+If the revision is successful, then the new event will be sent to
+`UpdateCommitEventWrapper` to be written to the database.
+Similar to creation, this function will require a user confirmation
+before it actually takes the action.
+
+## Delete Event
+
+Deletion is similar to update, except that you don't
+need to specify a constraint for the new event. If the user says
+"Delete my meeting with Emma", then the program is:
+```clojure
+(Yield :output 
+  (DeleteCommitEventWrapper :event 
+    (DeletePreflightEventWrapper :id 
+      (:id 
+        (singleton 
+          (:results 
+            (FindEventWrapperWithDefaults :constraint 
+              (Constraint[Event] :attendees 
+                (AttendeeListHasRecipientConstraint :recipientConstraint 
+                  (RecipientWithNameLike :name #(PersonName \"Emma\"))
+                )
+              )
+            )
+          )
+        )
+      )
+    )
+  )
+)
+```
+The program builds a `Constraint[Event]` specifying that the event's attendee
+list must contain "Emma". The constraint is sent to `FindEventWrapperWithDefaults`
+to query the database. If the event can
+be found, then its id is sent to `DeletePreflightEventWrapper` (which
+exists only to make the program's structure similar to update)
+and `DeleteCommitEventWrapper` (which performs the deletion). A user
+confirmation is required.
```

Comment:
Not sure if worth putting any footnotes on things that would ideally be different, but this would ideally use a salience call I think